doc/expressionLanguage.md

```diff
@@ -175,3 +176,35 @@ The following table shows expressions and their expected outcomes for a measure
 | '@value * 5.2'              | 31.2                  |
 | '"Pruebas " + "De Strings"' | 'Pruebas De Strings'  |
 | '@name value is @value'     | 'DevId629 value is 6' |
+
+## <a name="ngsiv2"/> NGSIv2 support
+
+As it is explained in previous sections, expressions can have two return types: String or Number, being the former one the default. Whenever a expression is executed without error, its result will be cast to the configured type. 
+
+On one hand, in NGSIv1 since all attributes'values are string, in the expression parser the expression type is set always to String and the transformation of the information coming from the SouthBound is done using replace instruction. Therefore, values sent to the CB will always be Strings. This can be seen in (#execution) example.
+
+On the other hand, NGSIv2 fully supports all the types described in the JSON
+specification (string, number, boolean, object, array and null). Therefore, the result of an expression must be cast to the appropriate type in order to avoid inconsistencies between the type field for an attribute and the type of the value that is being sent.
```

Comment:
Do "the appropiate type" refers to the type used to define the attribute? I mean, in the example shown in this documentation:

* The "appropiate type" for `location` would be String (actually it uses geo:point, but it is of string nature)
* The "appropiate type" for `temperature` and `fillingLevel` would be Number

or maybe "appropiate type" refers to the one associated to the expression being applied (so the type used to define the attribute is ignored in this case)?

Reply:
(This is actually related with this other comment https://github.com/dcalvoalonso/iotagent-node-lib/pull/3#discussion_r154142079... I'm just trying to fit all the pieces together in my mind and, in the process, trying to improve the tests and documentation :)

Reply:
Fixed in 776fe93a154bdd014f5bb7b37c8e165a52ab66d1